LDM-562.tex

```diff
@@ -9,9 +9,14 @@
 
 %% Retrieve date and model version
 \setDocUpstreamLocation{MagicDraw SysML}
-\setDocUpstreamVersion{44}
+\setDocUpstreamVersion{78}
```

Comment:
Should this also list a version of LDM-148? I don't know if the requirement-to-component mapping is defined in MagicDraw; if not, then to get the provenance of this document correct we'd need the LDM-148 version that the mapping was derived from. (If the LDM-148 mapping *is* derived from MagicDraw, it's not declared as such, which is an LDM-148 bug.)

Reply:
I haven't found the mapping from LSE-61 requirements to component but I believe that this has been done by @ktlim in MagicDraw. You make a good point though. I imagine it should be possible to link these requirements to the components in the model and then have the docgen specify the associated component so that it's explicit where the flow down came from.

cc/ @gcomoretto

Reply:
I don't believe that MagicDraw itself can tell me if I missed something though. It's possible that some ruby model validation code could be written to ensure that everything is tied together.

Reply:
The mapping from LSE-61 requirements to components is in MagicDraw as "Satisfy" relations in the DM Subsystem model.  These relationships are extracted as CSV and processed via a script to generate the LaTeX tables in the LDM-148 appendices.  The bulk of LDM-148 itself is not generated from MagicDraw, so "it's not declared as such" is not exactly an LDM-148 bug; I could try to record the provenance in the tables, however.

Gabriele and I talked yesterday about dividing up the components in such a way that (if done ideally) we can ensure that each component is covered by exactly one requirements document and test specification.